opm/grid/cpgrid/CartesianIndexMapper.hpp

```diff
@@ -49,6 +49,11 @@ namespace Dune
             return grid_.globalCell().size();
         }
 
+        int compressedLevelZeroSize() const
+        {
+            return (*(grid_.data_[0])).global_cell_.size();
```

Comment:
The problem here is that we need to use the correct member based on whether or not distributed_data_ is populated. Like above it works only if there is no entry in distributed_data_.
We need select either data_ or distributed_data_ depending on what current_view_data_ points to. Maybe we already have a function for that? If not we should probably implement one as there should be quite a few other occasion where we needed to the same thing (e.g. in .CpGrid::levelIndexSet)

Maybe using `size(0)` on CpGridData would be more expressive? Then we can skip one friend below.

Reply:
Oh I see, there are indeed a few occasions where we select between data_ and distributed_data_ in CpGrid implementation. I'll check if there is already a method doing that, so we replace those lines with it. 

size(0) gives total amount of cells on the CpGridData
global_cell_.size() gives total amount of ACTIVE cells on the CpGridData?

I was hesitating about this. Is this correct: compressed index == active index == value stored in global_cell_ [certain cell] ?

Reply:
we  (or at least I) are probably not really clear when we talk about this.
Only in the representation of the input file (EclipseGrid) inactive cells are represented. When constructing CpGrid from that information (after maybe even deactivating more), we simply skip the inactive cells. Information about inactive cells is available implicitly via global_cell_.

Hence the total amount of cells on CpGridData is the total amount of active cells. I guess compressed and active index are synonyms.

In global_cell_ the global (or cartesian index is stored). Hence for a 1x1x3 cartesian grid with the center cell inactive it would have 2 entries `{0 2}` and no entry would be equal to 1. (I had to use a 1D example because I always forget the ordering :sweat_smile:)

Reply:
Just saw that CpGrid::globalCell() internally distinguishes between data_ and distributed_data_ 
[https://github.com/OPM/opm-grid/blob/afe008a35ba4fb1b153826f651cb621552d05ca2/opm/grid/cpgrid/CpGrid.cpp#L604](url)